fuse/index.d.ts

```diff
@@ -1,30 +1,36 @@
 // Type definitions for Fuse.js 2.5.0
 // Project: https://github.com/krisk/Fuse
-// Definitions by: Greg Smith <https://github.com/smrq/>
+// Definitions by: Alan Agius <https://github.com/alan-agius4/>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
-declare class Fuse {
-	constructor(list: any[], options?: fuse.IFuseOptions);
-	search(pattern: string): any[];
-}
+declare module 'fuse.js' {
+	namespace fuse.js {
+		class Fuse {
+			constructor(list: any[], options?: FuseOptions);
+			search(pattern: string): any[];
+			search<T>(pattern: string): T[];
+		}
 
-declare namespace fuse {
-	interface IFuseOptions {
-		id?: string;
-		caseSensitive?: boolean;
-		include?: string[];
-		shouldSort?: boolean;
-		searchFn?: any;
-		sortFn?: (a: {score: number}, b: {score: number}) => number;
-		getFn?: (obj: any, path: string) => any;
-		keys?: string[] | { name:string; weight:number} [];
-        	verbose?:boolean;
-		tokenize?: boolean;
-		tokenSeparator? : RegExp;
-		matchAllTokens?: boolean;
-		location?: number;
-		distance?: number;
-		threshold?: number;
-		maxPatternLength?: number;
+		interface FuseOptions {
+			id?: string;
+			caseSensitive?: boolean;
+			include?: string[];
+			shouldSort?: boolean;
+			searchFn?: any;
+			sortFn?: (a: { score: number }, b: { score: number }) => number;
+			getFn?: (obj: any, path: string) => any;
+			keys?: string[] | { name: string; weight: number }[];
+			verbose?: boolean;
+			tokenize?: boolean;
+			tokenSeparator?: RegExp;
+			matchAllTokens?: boolean;
+			location?: number;
+			distance?: number;
+			threshold?: number;
+			maxPatternLength?: number;
+		}
 	}
+	export = fuse.js;
 }
+
+declare const Fuse: any;
```

Comment:
Looks like the package directly exports the `Fuse` class rather than exporting `fuse.Fuse`.
So the class should go outside of the namespace. Other things can still go in the namespace and this will be merged with the class.
Like this:

```ts
export = Fuse;
export as namespace Fuse;

declare class Fuse { ... }

declare namespace Fuse {
    interface FuseOptions { ... }
    interface SearchOptions { ... }
}
```